webproject/src/main/java/com/google/sps/services/implementations/JoinRoomRoomServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.google.sps.services.implementations;
+
+import javax.inject.Inject;
+
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.sps.authentication.AuthenticationHandler;
+import com.google.sps.data.UserRoom;
+import com.google.sps.proto.JoinRoomProto.JoinRoomResponse;
+import com.google.sps.proto.JoinRoomProto.JoinRoomRequest;
+import com.google.sps.services.interfaces.JoinRoomService;
+import com.google.sps.util.TimestampUtil;
+
+public class JoinRoomRoomServiceImpl implements JoinRoomService {
+    private AuthenticationHandler authenticationHandler;
+
+    @Inject
+    public JoinRoomRoomServiceImpl(AuthenticationHandler authenticationHandler) {
+        this.authenticationHandler = authenticationHandler;
+    }
+
+    @Override
+    public JoinRoomResponse execute(JoinRoomRequest postJoinRequest) {
+        String userEmail = authenticationHandler.getCurrentUser().getEmail();
+
+        FirebaseDatabase.getInstance()
```

Comment:
do you think we can move this to under UserRoom.java with function save?

Reply:
done. I am using UserRoomManager.java now :)